common/models/change.js

```diff
@@ -186,27 +186,51 @@ module.exports = function(Change) {
 
     cb = cb || utils.createPromiseCallback();
 
-    change.currentRevision(function(err, rev) {
+    var model = this.getModelCtor();
+    var id = this.getModelId();
+
+    model.findById(id, function(err, inst) {
       if (err) return cb(err);
 
-      // avoid setting rev and prev to the same value
-      if (currentRev === rev) {
-        change.debug('rev and prev are equal (not updating anything)');
-        return cb(null, change);
-      }
+      change.tenant = getTenant(model, inst);
```

Comment:
If I understand this part correctly, then it's adding a computed `tenant` property to each Change instance. I think it should be possible to accomplish this by using "before save" or "persist" hook.

```js
module.exports = myMixinFunction(TargetModel, config) {
  TargetModel.getChangeModel().observe('before save', function(ctx, next) {
    var tenant = getTenant(TargetModel, ctx.instance || ctx.data);
    if (tenant) {
     (ctx.instance || ctx.data).tenant = tenant;
    }
  });
});
```